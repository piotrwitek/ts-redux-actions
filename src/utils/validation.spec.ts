import {
  checkIsEmpty,
  checkInvalidActionCreator,
  checkInvalidActionType,
} from './validation';

/** HELPERS */

/** TESTS */

describe('checkIsEmpty', () => {
  describe('should pass when input is empty', () => {
    // @dts-jest:pass:snap
    checkIsEmpty(null); // => true
    // @dts-jest:pass:snap
    checkIsEmpty(undefined); // => true
  });
});

describe('checkInvalidActionCreator', () => {
  describe('should fail when input is valid action-creator', () => {
    const ac = () => ({ type: 'type' });
    ac.getType = () => 'type';
    // @dts-jest:pass:snap
    checkInvalidActionCreator(ac); // => false
  });
});

describe('checkInvalidActionType', () => {
  describe('should fail when input is valid action-type', () => {
    // @dts-jest:pass:snap
    checkInvalidActionType('asdf'); // => false
    // @dts-jest:pass:snap
    checkInvalidActionType(Symbol() as any); // => false
  });
});
