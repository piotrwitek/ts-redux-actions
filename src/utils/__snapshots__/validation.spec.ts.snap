// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`checkInvalidActionCreator(ac) 1`] = `"boolean"`;

exports[`checkInvalidActionType('asdf') 1`] = `"boolean"`;

exports[`checkInvalidActionType(Symbol() as any) 1`] = `"boolean"`;

exports[`checkIsEmpty(null) 1`] = `"boolean"`;

exports[`checkIsEmpty(undefined) 1`] = `"boolean"`;
